netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -198,6 +199,14 @@ private NettyTransportFactory(SocketAddress serverAddress,
       this.flowControlWindow = flowControlWindow;
       this.negotiator = negotiator;
       this.maxMessageSize = maxMessageSize;
+      if (serverAddress instanceof InetSocketAddress) {
```

Comment:
No. In general we don't want to be using InetSocketAddress like this because 1) it does I/O and 2) it conflicts with naming. Netty is a special case.